OpenCV CTU analysis crash

The results of an OpenCV cross-tu analysis:

histogram.cpp:
```
should always be wrapped in ArrayToPointerDecay
UNREACHABLE executed at /home/gamesh411/Projects/ericsson/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:2291!
#0 0x00000000012ccac4 PrintStackTraceSignalHandler(void*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x12ccac4)
#1 0x00000000012cce16 SignalHandler(int) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x12cce16)
#2 0x00007fa9e249d670 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x11670)
#3 0x00007fa9e100177f gsignal /build/glibc-mXZSwJ/glibc-2.24/signal/../sysdeps/unix/sysv/linux/raise.c:58:0
#4 0x00007fa9e100337a abort /build/glibc-mXZSwJ/glibc-2.24/stdlib/abort.c:91:0
#5 0x000000000128beeb (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x128beeb)
#6 0x000000000221a0c1 clang::ento::ExprEngine::VisitMemberExpr(clang::MemberExpr const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x221a0c1)
#7 0x0000000002214709 clang::ento::ExprEngine::Visit(clang::Stmt const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2214709)
#8 0x000000000220f8cf clang::ento::ExprEngine::ProcessStmt(clang::CFGStmt, clang::ento::ExplodedNode*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x220f8cf)
#9 0x000000000220f590 clang::ento::ExprEngine::processCFGElement(clang::CFGElement, clang::ento::ExplodedNode*, unsigned int, clang::ento::NodeBuilderContext*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x220f590)
#10 0x0000000002204d51 clang::ento::CoreEngine::HandlePostStmt(clang::CFGBlock const*, unsigned int, clang::ento::ExplodedNode*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2204d51)
#11 0x0000000002203f4f clang::ento::CoreEngine::ExecuteWorkList(clang::LocationContext const*, unsigned int, llvm::IntrusiveRefCntPtr<clang::ento::ProgramState const>) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2203f4f)
#12 0x0000000002015fb1 (anonymous namespace)::AnalysisConsumer::ActionExprEngine(clang::Decl*, bool, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2015fb1)
#13 0x0000000002015b5b (anonymous namespace)::AnalysisConsumer::HandleCode(clang::Decl*, unsigned int, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2015b5b)
#14 0x000000000200f022 (anonymous namespace)::AnalysisConsumer::HandleTranslationUnit(clang::ASTContext&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x200f022)
#15 0x00000000022ffc16 clang::ParseAST(clang::Sema&, bool, bool) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x22ffc16)
#16 0x000000000180d188 clang::FrontendAction::Execute() (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x180d188)
#17 0x00000000017d0f41 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x17d0f41)
#18 0x0000000001897040 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x1897040)
#19 0x000000000077ecad cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77ecad)
#20 0x000000000077d12a main (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77d12a)
#21 0x00007fa9e0fec3f1 __libc_start_main /build/glibc-mXZSwJ/glibc-2.24/csu/../csu/libc-start.c:325:0
#22 0x000000000077a5ca _start (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77a5ca)
Stack dump:
0.      Program arguments: /home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0 -cc1 -triple x86_64-unknown-linux-gnu -analyze -disable-free -main-file-name histogram.cpp -analyzer-store=region -analyzer-opt-analyze-nested-blocks -analyzer-eagerly-assume -analyzer-output plist -w -mrelocation-model pic -pic-level 2 -mthread-model posix -fmath-errno -masm-verbose -mconstructor-aliases -munwind-tables -fuse-init-array -target-cpu x86-64 -target-feature +sse -target-feature +sse2 -target-feature +sse3 -dwarf-column-info -debugger-tuning=gdb -ffunction-sections -coverage-notes-file /home/gamesh411/.codechecker/reports/clangsa_histogram.cpp_62dee806.gcno -resource-dir /home/gamesh411/Projects/ericsson/build/release/lib/clang/6.0.0 -D CVAPI_EXPORTS -D _USE_MATH_DEFINES -D __OPENCV_BUILD=1 -D __STDC_CONSTANT_MACROS -D __STDC_FORMAT_MACROS -D __STDC_LIMIT_MACROS -I /home/gamesh411/static_analysis/build/3rdparty/ippicv/ippicv_lnx/include -I /home/gamesh411/static_analysis/build/3rdparty/ippicv/ippiw_lnx/include -I /home/gamesh411/static_analysis/build -I /home/gamesh411/static_analysis/opencv/modules/imgproc/include -I /home/gamesh411/static_analysis/opencv/modules/imgproc/src -I /home/gamesh411/static_analysis/build/modules/imgproc -I /home/gamesh411/static_analysis/opencv/modules/core/include -I /usr/include/c++/6 -I /usr/include/x86_64-linux-gnu/c++/6 -I /usr/include/c++/6/backward -I /usr/local/include -I /home/gamesh411/Projects/ericsson/build/release/lib/clang/6.0.0/include -I /usr/include/x86_64-linux-gnu -I /usr/include -D DEBUG -D _DEBUG -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/x86_64-linux-gnu/c++/6.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/x86_64-linux-gnu/c++/6.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/backward -internal-isystem /usr/local/include -internal-isystem /home/gamesh411/Projects/ericsson/build/release/lib/clang/6.0.0/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -Wno-long-long -fdeprecated-macro -fdebug-compilation-dir /home/gamesh411/static_analysis/build -ferror-limit 19 -fmessage-length 0 -fvisibility hidden -fvisibility-inlines-hidden -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -analyzer-opt-analyze-headers -analyzer-output=plist-multi-file -analyzer-disable-checker alpha.clone.CloneChecker -analyzer-disable-checker alpha.core.BoolAssignment -analyzer-disable-checker alpha.core.CallAndMessageUnInitRefArg -analyzer-disable-checker alpha.core.CastSize -analyzer-disable-checker alpha.core.CastToStruct -analyzer-disable-checker alpha.core.Conversion -analyzer-disable-checker alpha.core.DynamicTypeChecker -analyzer-disable-checker alpha.core.FixedAddr -analyzer-disable-checker alpha.core.IdenticalExpr -analyzer-disable-checker alpha.core.PointerArithm -analyzer-disable-checker alpha.core.PointerSub -analyzer-disable-checker alpha.core.SizeofPtr -analyzer-disable-checker alpha.core.TestAfterDivZero -analyzer-disable-checker alpha.cplusplus.IteratorRange -analyzer-disable-checker alpha.cplusplus.MisusedMovedObject -analyzer-disable-checker alpha.deadcode.UnreachableCode -analyzer-disable-checker alpha.osx.cocoa.DirectIvarAssignment -analyzer-disable-checker alpha.osx.cocoa.DirectIvarAssignmentForAnnotatedFunctions -analyzer-disable-checker alpha.osx.cocoa.InstanceVariableInvalidation -analyzer-disable-checker alpha.osx.cocoa.MissingInvalidationMethod -analyzer-disable-checker alpha.osx.cocoa.localizability.PluralMisuseChecker -analyzer-disable-checker alpha.security.ArrayBound -analyzer-disable-checker alpha.security.ArrayBoundV2 -analyzer-disable-checker alpha.security.MallocOverflow -analyzer-disable-checker alpha.security.ReturnPtrRange -analyzer-disable-checker alpha.security.taint.TaintPropagation -analyzer-disable-checker alpha.unix.BlockInCriticalSection -analyzer-disable-checker alpha.unix.Chroot -analyzer-disable-checker alpha.unix.PthreadLock -analyzer-disable-checker alpha.unix.SimpleStream -analyzer-disable-checker alpha.unix.Stream -analyzer-disable-checker alpha.unix.cstring.BufferOverlap -analyzer-disable-checker alpha.unix.cstring.NotNullTerminated -analyzer-disable-checker alpha.unix.cstring.OutOfBounds -analyzer-disable-checker apiModeling.google.GTest -analyzer-checker=core.CallAndMessage -analyzer-checker=core.DivideZero -analyzer-checker=core.DynamicTypePropagation -analyzer-checker=core.NonNullParamChecker -analyzer-checker=core.NullDereference -analyzer-checker=core.StackAddressEscape -analyzer-checker=core.UndefinedBinaryOperatorResult -analyzer-checker=core.VLASize -analyzer-checker=core.builtin.BuiltinFunctions -analyzer-checker=core.builtin.NoReturnFunctions -analyzer-checker=core.uninitialized.ArraySubscript -analyzer-checker=core.uninitialized.Assign -analyzer-checker=core.uninitialized.Branch -analyzer-checker=core.uninitialized.CapturedBlockVariable -analyzer-checker=core.uninitialized.UndefReturn -analyzer-checker=cplusplus.NewDelete -analyzer-checker=cplusplus.NewDeleteLeaks -analyzer-checker=cplusplus.SelfAssignment -analyzer-checker=deadcode.DeadStores -analyzer-disable-checker debug.AnalysisOrder -analyzer-disable-checker debug.ConfigDumper -analyzer-disable-checker debug.DumpBugHash -analyzer-disable-checker debug.DumpCFG -analyzer-disable-checker debug.DumpCallGraph -analyzer-disable-checker debug.DumpCalls -analyzer-disable-checker debug.DumpDominators -analyzer-disable-checker debug.DumpLiveVars -analyzer-disable-checker debug.DumpTraversal -analyzer-disable-checker debug.ExprInspection -analyzer-disable-checker debug.Stats -analyzer-disable-checker debug.TaintTest -analyzer-disable-checker debug.ViewCFG -analyzer-disable-checker debug.ViewCallGraph -analyzer-disable-checker debug.ViewExplodedGraph -analyzer-disable-checker llvm.Conventions -analyzer-disable-checker nullability.NullPassedToNonnull -analyzer-disable-checker nullability.NullReturnedFromNonnull -analyzer-disable-checker nullability.NullableDereferenced -analyzer-disable-checker nullability.NullablePassedToNonnull -analyzer-disable-checker nullability.NullableReturnedFromNonnull -analyzer-disable-checker optin.cplusplus.VirtualCall -analyzer-disable-checker optin.mpi.MPI-Checker -analyzer-disable-checker optin.osx.cocoa.localizability.EmptyLocalizationContextChecker -analyzer-disable-checker optin.osx.cocoa.localizability.NonLocalizedStringChecker -analyzer-disable-checker optin.performance.Padding -analyzer-disable-checker optin.portability.UnixAPI -analyzer-disable-checker osx.API -analyzer-disable-checker osx.NumberObjectConversion -analyzer-disable-checker osx.ObjCProperty -analyzer-disable-checker osx.SecKeychainAPI -analyzer-disable-checker osx.cocoa.AtSync -analyzer-disable-checker osx.cocoa.ClassRelease -analyzer-disable-checker osx.cocoa.Dealloc -analyzer-disable-checker osx.cocoa.IncompatibleMethodTypes -analyzer-disable-checker osx.cocoa.Loops -analyzer-disable-checker osx.cocoa.MissingSuperCall -analyzer-disable-checker osx.cocoa.NSAutoreleasePool -analyzer-disable-checker osx.cocoa.NSError -analyzer-disable-checker osx.cocoa.NilArg -analyzer-disable-checker osx.cocoa.NonNilReturnValue -analyzer-disable-checker osx.cocoa.ObjCGenerics -analyzer-disable-checker osx.cocoa.RetainCount -analyzer-disable-checker osx.cocoa.SelfInit -analyzer-disable-checker osx.cocoa.SuperDealloc -analyzer-disable-checker osx.cocoa.UnusedIvars -analyzer-disable-checker osx.cocoa.VariadicMethodTypes -analyzer-disable-checker osx.coreFoundation.CFError -analyzer-disable-checker osx.coreFoundation.CFNumber -analyzer-disable-checker osx.coreFoundation.CFRetainRelease -analyzer-disable-checker osx.coreFoundation.containers.OutOfBounds -analyzer-disable-checker osx.coreFoundation.containers.PointerSizedValues -analyzer-disable-checker security.FloatLoopCounter -analyzer-checker=security.insecureAPI.UncheckedReturn -analyzer-checker=security.insecureAPI.getpw -analyzer-checker=security.insecureAPI.gets -analyzer-checker=security.insecureAPI.mkstemp -analyzer-checker=security.insecureAPI.mktemp -analyzer-disable-checker security.insecureAPI.rand -analyzer-disable-checker security.insecureAPI.strcpy -analyzer-checker=security.insecureAPI.vfork -analyzer-checker=unix.API -analyzer-checker=unix.Malloc -analyzer-checker=unix.MallocSizeof -analyzer-checker=unix.MismatchedDeallocator -analyzer-checker=unix.StdCLibraryFunctions -analyzer-checker=unix.Vfork -analyzer-checker=unix.cstring.BadSizeArg -analyzer-checker=unix.cstring.NullArg -analyzer-disable-checker valist.CopyToSelf -analyzer-disable-checker valist.Uninitialized -analyzer-disable-checker valist.Unterminated -o /home/gamesh411/.codechecker/reports/clangsa_histogram.cpp_62dee806.plist -x c++ /home/gamesh411/static_analysis/opencv/modules/imgproc/src/histogram.cpp 
1.      <eof> parser at end of file
2.      While analyzing stack: 
        #0 void cvNormalizeHist(CvHistogram *hist, double factor) __attribute__((visibility("default")))
        #1 void cvCalcArrBackProjectPatch(CvArr **arr, CvArr *dst, CvSize patch_size, CvHistogram *hist, int method, double norm_factor) __attribute__((visibility("default")))
3.      /home/gamesh411/static_analysis/opencv/modules/imgproc/src/histogram.cpp:2981:15: Error evaluating statement
4.      /home/gamesh411/static_analysis/opencv/modules/imgproc/src/histogram.cpp:2981:15: Error evaluating statement
clang-6.0: error: unable to execute command: Aborted (core dumped)
clang-6.0: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 6.0.0 (https://github.com/Ericsson/clang 4b38271e218709c82a843ba340e3cdbabbdfd8f2) (https://github.com/llvm-mirror/llvm 0472b1ccd49cc2737cc35037829371eb9cdbc07c)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/gamesh411/ctu_release/bin
clang-6.0: note: diagnostic msg: PLEASE submit a bug report to http://llvm.org/bugs/ and include the crash backtrace, preprocessed source, and associated run script.
clang-6.0: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-6.0: note: diagnostic msg: /tmp/histogram-5fcbcd.cpp
clang-6.0: note: diagnostic msg: /tmp/histogram-5fcbcd.sh
clang-6.0: note: diagnostic msg:
```

The immediate cause of the crash is the same for the other 2 files:
quadsubpix.cpp:
```
should always be wrapped in ArrayToPointerDecay
UNREACHABLE executed at /home/gamesh411/Projects/ericsson/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:2291!
#0 0x00000000012ccac4 PrintStackTraceSignalHandler(void*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x12ccac4)
#1 0x00000000012cce16 SignalHandler(int) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x12cce16)
#2 0x00007f60d0b9b670 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x11670)
#3 0x00007f60cf6ff77f gsignal /build/glibc-mXZSwJ/glibc-2.24/signal/../sysdeps/unix/sysv/linux/raise.c:58:0
#4 0x00007f60cf70137a abort /build/glibc-mXZSwJ/glibc-2.24/stdlib/abort.c:91:0
#5 0x000000000128beeb (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x128beeb)
#6 0x000000000221a0c1 clang::ento::ExprEngine::VisitMemberExpr(clang::MemberExpr const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x221a0c1)
#7 0x0000000002214709 clang::ento::ExprEngine::Visit(clang::Stmt const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2214709)
#8 0x000000000220f8cf clang::ento::ExprEngine::ProcessStmt(clang::CFGStmt, clang::ento::ExplodedNode*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x220f8cf)
#9 0x000000000220f590 clang::ento::ExprEngine::processCFGElement(clang::CFGElement, clang::ento::ExplodedNode*, unsigned int, clang::ento::NodeBuilderContext*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x220f590)
#10 0x0000000002204d51 clang::ento::CoreEngine::HandlePostStmt(clang::CFGBlock const*, unsigned int, clang::ento::ExplodedNode*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2204d51)
#11 0x0000000002203f4f clang::ento::CoreEngine::ExecuteWorkList(clang::LocationContext const*, unsigned int, llvm::IntrusiveRefCntPtr<clang::ento::ProgramState const>) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2203f4f)
#12 0x0000000002015fb1 (anonymous namespace)::AnalysisConsumer::ActionExprEngine(clang::Decl*, bool, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2015fb1)
#13 0x0000000002015b5b (anonymous namespace)::AnalysisConsumer::HandleCode(clang::Decl*, unsigned int, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2015b5b)
#14 0x000000000200f022 (anonymous namespace)::AnalysisConsumer::HandleTranslationUnit(clang::ASTContext&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x200f022)
#15 0x00000000022ffc16 clang::ParseAST(clang::Sema&, bool, bool) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x22ffc16)
#16 0x000000000180d188 clang::FrontendAction::Execute() (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x180d188)
#17 0x00000000017d0f41 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x17d0f41)
#18 0x0000000001897040 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x1897040)
#19 0x000000000077ecad cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77ecad)
#20 0x000000000077d12a main (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77d12a)
#21 0x00007f60cf6ea3f1 __libc_start_main /build/glibc-mXZSwJ/glibc-2.24/csu/../csu/libc-start.c:325:0
#22 0x000000000077a5ca _start (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77a5ca)
```

```
1.      <eof> parser at end of file
2.      While analyzing stack: 
        #0 static int segment_hist_max(const cv::Mat &hist, int &low_thresh, int &high_thresh)
        #1 bool cv::find4QuadCornerSubpix(cv::InputArray _img, cv::InputOutputArray _corners, cv::Size region_size) __attribute__((visibility("default")))
3.      /home/gamesh411/static_analysis/opencv/modules/calib3d/src/quadsubpix.cpp:121:24: Error evaluating statement
4.      /home/gamesh411/static_analysis/opencv/modules/calib3d/src/quadsubpix.cpp:121:24: Error evaluating statement
```

and test_fundam.cpp:
```
should always be wrapped in ArrayToPointerDecay
UNREACHABLE executed at /home/gamesh411/Projects/ericsson/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:2291!
#0 0x00000000012ccac4 PrintStackTraceSignalHandler(void*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x12ccac4)
#1 0x00000000012cce16 SignalHandler(int) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x12cce16)
#2 0x00007f9eac073670 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x11670)
#3 0x00007f9eaabd777f gsignal /build/glibc-mXZSwJ/glibc-2.24/signal/../sysdeps/unix/sysv/linux/raise.c:58:0
#4 0x00007f9eaabd937a abort /build/glibc-mXZSwJ/glibc-2.24/stdlib/abort.c:91:0
#5 0x000000000128beeb (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x128beeb)
#6 0x000000000221a0c1 clang::ento::ExprEngine::VisitMemberExpr(clang::MemberExpr const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x221a0c1)
#7 0x0000000002214709 clang::ento::ExprEngine::Visit(clang::Stmt const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2214709)
#8 0x000000000220f8cf clang::ento::ExprEngine::ProcessStmt(clang::CFGStmt, clang::ento::ExplodedNode*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x220f8cf)
#9 0x000000000220f590 clang::ento::ExprEngine::processCFGElement(clang::CFGElement, clang::ento::ExplodedNode*, unsigned int, clang::ento::NodeBuilderContext*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x220f590)
#10 0x0000000002204d51 clang::ento::CoreEngine::HandlePostStmt(clang::CFGBlock const*, unsigned int, clang::ento::ExplodedNode*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2204d51)
#11 0x0000000002203f4f clang::ento::CoreEngine::ExecuteWorkList(clang::LocationContext const*, unsigned int, llvm::IntrusiveRefCntPtr<clang::ento::ProgramState const>) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2203f4f)
#12 0x0000000002015fb1 (anonymous namespace)::AnalysisConsumer::ActionExprEngine(clang::Decl*, bool, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2015fb1)
#13 0x0000000002015b5b (anonymous namespace)::AnalysisConsumer::HandleCode(clang::Decl*, unsigned int, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x2015b5b)
#14 0x000000000200f022 (anonymous namespace)::AnalysisConsumer::HandleTranslationUnit(clang::ASTContext&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x200f022)
#15 0x00000000022ffc16 clang::ParseAST(clang::Sema&, bool, bool) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x22ffc16)
#16 0x000000000180d188 clang::FrontendAction::Execute() (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x180d188)
#17 0x00000000017d0f41 clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x17d0f41)
#18 0x0000000001897040 clang::ExecuteCompilerInvocation(clang::CompilerInstance*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x1897040)
#19 0x000000000077ecad cc1_main(llvm::ArrayRef<char const*>, char const*, void*) (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77ecad)
#20 0x000000000077d12a main (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77d12a)
#21 0x00007f9eaabc23f1 __libc_start_main /build/glibc-mXZSwJ/glibc-2.24/csu/../csu/libc-start.c:325:0
#22 0x000000000077a5ca _start (/home/gamesh411/Projects/ericsson/build/release/bin/clang-6.0+0x77a5ca)
```

```
1.      <eof> parser at end of file
2.      While analyzing stack: 
        #0 int cvTsRodrigues(const CvMat *src, CvMat *dst, CvMat *jacobian)
        #1 void cvtest::Rodrigues(const cv::Mat &src, cv::Mat &dst, cv::Mat *jac = 0)
        #2 void CV_EssentialMatTest::fill_array(int test_case_idx, int i, int j, cv::Mat &arr)
3.      /home/gamesh411/static_analysis/opencv/modules/calib3d/test/test_fundam.cpp:231:15: Error evaluating statement
4.      /home/gamesh411/static_analysis/opencv/modules/calib3d/test/test_fundam.cpp:231:15: Error evaluating statement
```

Another issue came up for the histogram.cpp. I made many runs of the analyzer in CTU mode. The error that i have received earlier came up again when the CodeChecker was run in --ctu-all mode (i guess this means the mapping files get deleted after each run, so afterwards if i just ran the analyzer-command again, the needed files were missing).

The same histogram.cpp file showed the following stacktrace, leading me to belive that the issue is not some array-related expression not getting wrapped in decay-to-pointer, rather that the importer somehow decided that the definition imported was somehow not equivalent to itself.

```
clang-5.0: /home/gamesh411/llvm/tools/clang/lib/AST/DeclBase.cpp:1328: void clang::DeclContext::removeDecl(clang::Decl *): Assertion `(D->NextInContextAndBits.getPointer() || D == LastDecl) && "decl is not in decls list"' failed.
#0 0x0000000002166469 llvm::sys::PrintStackTrace(llvm::raw_ostream&) /home/gamesh411/llvm/lib/Support/Unix/Signals.inc:398:11
#1 0x0000000002166619 PrintStackTraceSignalHandler(void*) /home/gamesh411/llvm/lib/Support/Unix/Signals.inc:462:1
#2 0x0000000002164c73 llvm::sys::RunSignalHandlers() /home/gamesh411/llvm/lib/Support/Signals.cpp:0:5
#3 0x0000000002166974 SignalHandler(int) /home/gamesh411/llvm/lib/Support/Unix/Signals.inc:252:1
#4 0x00007fded2ef7670 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x11670)
#5 0x00007fded1a5b77f gsignal /build/glibc-mXZSwJ/glibc-2.24/signal/../sysdeps/unix/sysv/linux/raise.c:58:0
#6 0x00007fded1a5d37a abort /build/glibc-mXZSwJ/glibc-2.24/stdlib/abort.c:91:0
#7 0x00007fded1a53b47 __assert_fail_base /build/glibc-mXZSwJ/glibc-2.24/assert/assert.c:92:0
#8 0x00007fded1a53bf2 (/lib/x86_64-linux-gnu/libc.so.6+0x2dbf2)
#9 0x0000000005a9064e clang::DeclContext::removeDecl(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/DeclBase.cpp:1331:7
#10 0x00000000059f22ee clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:997:5
#11 0x00000000059f1c60 clang::ASTNodeImporter::ImportDefinition(clang::RecordDecl*, clang::RecordDecl*, clang::ASTNodeImporter::ImportDefinitionKind) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1119:3
#12 0x00000000059f7b7c clang::ASTNodeImporter::VisitRecordDecl(clang::RecordDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1980:7
#13 0x0000000005a1fdd8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::VisitCXXRecordDecl(clang::CXXRecordDecl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:251:1
#14 0x0000000005a18990 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:251:1
#15 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#16 0x00000000059f048e clang::ASTImporter::ImportContext(clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6352:23
#17 0x00000000059f02b2 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:900:3
#18 0x00000000059f843c clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:2051:7
#19 0x00000000059fa55d clang::ASTNodeImporter::VisitCXXMethodDecl(clang::CXXMethodDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:2265:3
#20 0x00000000059fa58d clang::ASTNodeImporter::VisitCXXConstructorDecl(clang::CXXConstructorDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:2269:3
#21 0x0000000005a18bc0 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:403:1
#22 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#23 0x00000000059f2217 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:994:30
#24 0x00000000059f4906 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1540:10
#25 0x0000000005a18798 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#26 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#27 0x00000000059f048e clang::ASTImporter::ImportContext(clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6352:23
#28 0x00000000059f02b2 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:900:3
#29 0x00000000059f422c clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1478:7
#30 0x0000000005a18798 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#31 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#32 0x00000000059f048e clang::ASTImporter::ImportContext(clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6352:23
#33 0x00000000059f02b2 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:900:3
#34 0x00000000059f6823 clang::ASTNodeImporter::VisitRecordDecl(clang::RecordDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1797:7
#35 0x0000000005a1fdd8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::VisitCXXRecordDecl(clang::CXXRecordDecl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:251:1
#36 0x0000000005a18990 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:251:1
#37 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#38 0x00000000059f6748 clang::ASTNodeImporter::VisitRecordDecl(clang::RecordDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1785:11
#39 0x0000000005a1fdd8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::VisitCXXRecordDecl(clang::CXXRecordDecl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:251:1
#40 0x0000000005a18990 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:251:1
#41 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#42 0x00000000059f2217 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:994:30
#43 0x00000000059f4906 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1540:10
#44 0x0000000005a18798 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#45 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#46 0x00000000059f2217 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:994:30
#47 0x00000000059f4906 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1540:10
#48 0x0000000005a18798 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#49 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#50 0x00000000059f048e clang::ASTImporter::ImportContext(clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6352:23
#51 0x00000000059f02b2 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:900:3
#52 0x00000000059f843c clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:2051:7
#53 0x0000000005a18b6c clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:389:1
#54 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#55 0x0000000005a12dda clang::ASTNodeImporter::VisitUnresolvedLookupExpr(clang::UnresolvedLookupExpr*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:5956:26
#56 0x0000000005a1a46f clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/StmtNodes.inc:721:1
#57 0x00000000059f5c13 clang::ASTImporter::Import(clang::Stmt*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6421:9
#58 0x00000000059ed8c5 clang::ASTImporter::Import(clang::Expr*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6407:10
#59 0x0000000005a13dd1 clang::ASTNodeImporter::VisitCallExpr(clang::CallExpr*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6070:9
#60 0x0000000005a21268 clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::VisitCXXOperatorCallExpr(clang::CXXOperatorCallExpr*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/StmtNodes.inc:345:1
#61 0x0000000005a19e4f clang::StmtVisitorBase<clang::make_ptr, clang::ASTNodeImporter, clang::Stmt*>::Visit(clang::Stmt*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/StmtNodes.inc:345:1
#62 0x00000000059f5c13 clang::ASTImporter::Import(clang::Stmt*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6421:9
#63 0x00000000059ed8c5 clang::ASTImporter::Import(clang::Expr*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6407:10
#64 0x00000000059f821c clang::ASTNodeImporter::VisitEnumConstantDecl(clang::EnumConstantDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:2030:9
#65 0x0000000005a18d10 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:491:1
#66 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#67 0x00000000059f2217 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:994:30
#68 0x00000000059f1e45 clang::ASTNodeImporter::ImportDefinition(clang::EnumDecl*, clang::EnumDecl*, clang::ASTNodeImporter::ImportDefinitionKind) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1165:3
#69 0x00000000059f6694 clang::ASTNodeImporter::VisitEnumDecl(clang::EnumDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1773:7
#70 0x0000000005a18958 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:241:1
#71 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#72 0x00000000059f2217 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:994:30
#73 0x00000000059f1c60 clang::ASTNodeImporter::ImportDefinition(clang::RecordDecl*, clang::RecordDecl*, clang::ASTNodeImporter::ImportDefinitionKind) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1119:3
#74 0x0000000005a057d3 clang::ASTNodeImporter::VisitClassTemplateSpecializationDecl(clang::ClassTemplateSpecializationDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:4030:7
#75 0x0000000005a1fe08 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::VisitClassTemplatePartialSpecializationDecl(clang::ClassTemplatePartialSpecializationDecl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:259:1
#76 0x0000000005a189c8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:259:1
#77 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#78 0x00000000059f2217 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:994:30
#79 0x00000000059f4906 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1540:10
#80 0x0000000005a18798 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#81 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#82 0x00000000059f048e clang::ASTImporter::ImportContext(clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6352:23
#83 0x00000000059f02b2 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:900:3
#84 0x00000000059f4d79 clang::ASTNodeImporter::VisitTypedefNameDecl(clang::TypedefNameDecl*, bool) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1587:7
#85 0x00000000059f57ef clang::ASTNodeImporter::VisitTypedefDecl(clang::TypedefDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1663:3
#86 0x0000000005a18a38 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:303:1
#87 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#88 0x00000000059ee5a7 clang::ASTNodeImporter::VisitTypedefType(clang::TypedefType const*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:635:16
#89 0x0000000005a18294 clang::TypeVisitor<clang::ASTNodeImporter, clang::QualType>::Visit(clang::Type const*) /home/gamesh411/llvm/tools/clang/include/clang/AST/TypeNodes.def:83:1
#90 0x00000000059ec4b6 clang::ASTImporter::Import(clang::QualType) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6266:27
#91 0x00000000059eddc2 clang::ASTNodeImporter::VisitFunctionProtoType(clang::FunctionProtoType const*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:574:33
#92 0x0000000005a18224 clang::TypeVisitor<clang::ASTNodeImporter, clang::QualType>::Visit(clang::Type const*) /home/gamesh411/llvm/tools/clang/include/clang/AST/TypeNodes.def:79:1
#93 0x00000000059ec4b6 clang::ASTImporter::Import(clang::QualType) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6266:27
#94 0x00000000059f8df1 clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:2138:25
#95 0x0000000005a18b6c clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:389:1
#96 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#97 0x0000000005a06fe6 clang::ASTNodeImporter::VisitFunctionTemplateDecl(clang::FunctionTemplateDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:4279:31
#98 0x0000000005a188e8 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:201:1
#99 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#100 0x00000000059f2217 clang::ASTNodeImporter::ImportDeclContext(clang::DeclContext*, bool, clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:994:30
#101 0x00000000059f4906 clang::ASTNodeImporter::VisitNamespaceDecl(clang::NamespaceDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:1540:10
#102 0x0000000005a18798 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:105:1
#103 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#104 0x00000000059f048e clang::ASTImporter::ImportContext(clang::DeclContext*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6352:23
#105 0x00000000059f0330 clang::ASTNodeImporter::ImportDeclParts(clang::NamedDecl*, clang::DeclContext*&, clang::DeclContext*&, clang::DeclarationName&, clang::NamedDecl*&, clang::SourceLocation&) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:906:5
#106 0x00000000059f843c clang::ASTNodeImporter::VisitFunctionDecl(clang::FunctionDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:2051:7
#107 0x00000000059fa55d clang::ASTNodeImporter::VisitCXXMethodDecl(clang::CXXMethodDecl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:2265:3
#108 0x0000000005a18ba4 clang::declvisitor::Base<clang::declvisitor::make_ptr, clang::ASTNodeImporter, clang::Decl*>::Visit(clang::Decl*) /home/gamesh411/ericsson_clang_debug/tools/clang/include/clang/AST/DeclNodes.inc:399:1
#109 0x00000000059ee1ca clang::ASTImporter::Import(clang::Decl*) /home/gamesh411/llvm/tools/clang/lib/AST/ASTImporter.cpp:6316:9
#110 0x00000000047b5c0e clang::tooling::CrossTranslationUnit::getCrossTUDefinition(clang::FunctionDecl const*, llvm::StringRef, llvm::StringRef, llvm::StringRef) /home/gamesh411/llvm/tools/clang/lib/Tooling/CrossTranslationUnit.cpp:196:20
#111 0x00000000046b8eb8 clang::ento::AnyFunctionCall::getRuntimeDefinition() const /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/CallEvent.cpp:384:37
#112 0x00000000046b9f3a clang::ento::CXXInstanceCall::getRuntimeDefinition() const /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/CallEvent.cpp:522:29
#113 0x00000000046ba4ac clang::ento::CXXMemberCall::getRuntimeDefinition() const /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/CallEvent.cpp:621:27
#114 0x000000000472e206 clang::ento::ExprEngine::defaultEvalCall(clang::ento::NodeBuilder&, clang::ento::ExplodedNode*, clang::ento::CallEvent const&) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngineCallAndReturn.cpp:915:34
#115 0x00000000046c7dbf clang::ento::CheckerManager::runCheckersForEvalCall(clang::ento::ExplodedNodeSet&, clang::ento::ExplodedNodeSet const&, clang::ento::CallEvent const&, clang::ento::ExprEngine&) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/CheckerManager.cpp:619:11
#116 0x000000000472d039 clang::ento::ExprEngine::evalCall(clang::ento::ExplodedNodeSet&, clang::ento::ExplodedNode*, clang::ento::CallEvent const&) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngineCallAndReturn.cpp:521:3
#117 0x000000000472cefb clang::ento::ExprEngine::VisitCallExpr(clang::CallExpr const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngineCallAndReturn.cpp:0:5
#118 0x00000000046fab64 clang::ento::ExprEngine::Visit(clang::Stmt const*, clang::ento::ExplodedNode*, clang::ento::ExplodedNodeSet&) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:1232:7
#119 0x00000000046f7ce4 clang::ento::ExprEngine::ProcessStmt(clang::CFGStmt, clang::ento::ExplodedNode*) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:547:5
#120 0x00000000046f79aa clang::ento::ExprEngine::processCFGElement(clang::CFGElement, clang::ento::ExplodedNode*, unsigned int, clang::ento::NodeBuilderContext*) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/ExprEngine.cpp:394:7
#121 0x00000000046e2b06 clang::ento::CoreEngine::HandlePostStmt(clang::CFGBlock const*, unsigned int, clang::ento::ExplodedNode*) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/CoreEngine.cpp:532:1
#122 0x00000000046e2403 clang::ento::CoreEngine::dispatchWorkItem(clang::ento::ExplodedNode*, clang::ProgramPoint, clang::ento::WorkListUnit const&) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/CoreEngine.cpp:281:1
#123 0x00000000046e1f98 clang::ento::CoreEngine::ExecuteWorkList(clang::LocationContext const*, unsigned int, llvm::IntrusiveRefCntPtr<clang::ento::ProgramState const>) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Core/CoreEngine.cpp:216:3
#124 0x00000000040d7335 clang::ento::ExprEngine::ExecuteWorkList(clang::LocationContext const*, unsigned int) /home/gamesh411/llvm/tools/clang/include/clang/StaticAnalyzer/Core/PathSensitive/ExprEngine.h:114:19
#125 0x000000000408217b (anonymous namespace)::AnalysisConsumer::ActionExprEngine(clang::Decl*, bool, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:741:7
#126 0x0000000004081f11 (anonymous namespace)::AnalysisConsumer::RunPathSensitiveChecks(clang::Decl*, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:762:5
#127 0x0000000004081bc5 (anonymous namespace)::AnalysisConsumer::HandleCode(clang::Decl*, unsigned int, clang::ento::ExprEngine::InliningModes, llvm::DenseSet<clang::Decl const*, llvm::DenseMapInfo<clang::Decl const*> >*) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:709:15
#128 0x000000000407532a (anonymous namespace)::AnalysisConsumer::HandleDeclsCallGraph(unsigned int) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:521:5
#129 0x000000000407358d (anonymous namespace)::AnalysisConsumer::HandleTranslationUnit(clang::ASTContext&) /home/gamesh411/llvm/tools/clang/lib/StaticAnalyzer/Frontend/AnalysisConsumer.cpp:576:5
#130 0x00000000048dcf66 clang::ParseAST(clang::Sema&, bool, bool) /home/gamesh411/llvm/tools/clang/lib/Parse/ParseAST.cpp:159:13
#131 0x0000000002b7df8c clang::ASTFrontendAction::ExecuteAction() /home/gamesh411/llvm/tools/clang/lib/Frontend/FrontendAction.cpp:1005:1
#132 0x0000000002b7da00 clang::FrontendAction::Execute() /home/gamesh411/llvm/tools/clang/lib/Frontend/FrontendAction.cpp:906:7
#133 0x0000000002b0d90a clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) /home/gamesh411/llvm/tools/clang/lib/Frontend/CompilerInstance.cpp:981:7
#134 0x0000000002cf3adf clang::ExecuteCompilerInvocation(clang::CompilerInstance*) /home/gamesh411/llvm/tools/clang/lib/FrontendTool/ExecuteCompilerInvocation.cpp:251:8
#135 0x0000000000d7a499 cc1_main(llvm::ArrayRef<char const*>, char const*, void*) /home/gamesh411/llvm/tools/clang/tools/driver/cc1_main.cpp:221:11
#136 0x0000000000d6d669 ExecuteCC1Tool(llvm::ArrayRef<char const*>, llvm::StringRef) /home/gamesh411/llvm/tools/clang/tools/driver/driver.cpp:306:5
#137 0x0000000000d6c2a0 main /home/gamesh411/llvm/tools/clang/tools/driver/driver.cpp:387:5
#138 0x00007fded1a463f1 __libc_start_main /build/glibc-mXZSwJ/glibc-2.24/csu/../csu/libc-start.c:325:0
#139 0x0000000000d6b26a _start (/home/gamesh411/ericsson_clang_debug/bin/clang-5.0+0xd6b26a)
Stack dump:
0.      Program arguments: /home/gamesh411/ericsson_clang_debug/bin/clang-5.0 -cc1 -triple x86_64-unknown-linux-gnu -analyze -disable-free -main-file-name histogram.cpp -analyzer-store=region -analyzer-opt-analyze-nested-blocks -analyzer-eagerly-assume -analyzer-output plist -w -mrelocation-model pic -pic-level 2 -mthread-model posix -fmath-errno -masm-verbose -mconstructor-aliases -munwind-tables -fuse-init-array -target-cpu x86-64 -target-feature +sse -target-feature +sse2 -target-feature +sse3 -dwarf-column-info -debugger-tuning=gdb -ffunction-sections -coverage-notes-file /home/gamesh411/build_opencv3_git/opencvgcc/histogram.cpp_5f1f94776187c199b4162da78b6f89f3.gcno -resource-dir /home/gamesh411/ericsson_clang_debug/lib/clang/5.0.0 -isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0 -isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/x86_64-linux-gnu/c++/6.3.0 -isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/backward -isystem /usr/local/include -isystem /home/gamesh411/ericsson_clang_release/lib/clang/5.0.0/include -isystem /usr/include/x86_64-linux-gnu -isystem /usr/include -D CVAPI_EXPORTS -D _USE_MATH_DEFINES -D __OPENCV_BUILD=1 -D __STDC_CONSTANT_MACROS -D __STDC_FORMAT_MACROS -D __STDC_LIMIT_MACROS -I /home/gamesh411/build_opencv3_git/3rdparty/ippicv/ippicv_lnx/include -I /home/gamesh411/build_opencv3_git/3rdparty/ippicv/ippiw_lnx/include -I /home/gamesh411/build_opencv3_git -I /home/gamesh411/opencv-3.3.0/modules/imgproc/include -I /home/gamesh411/opencv-3.3.0/modules/imgproc/src -I /home/gamesh411/build_opencv3_git/modules/imgproc -I /home/gamesh411/opencv-3.3.0/modules/core/include -I /usr/include/c++/6 -I /usr/include/x86_64-linux-gnu/c++/6 -I /usr/include/c++/6/backward -I /usr/local/include -I /home/gamesh411/ericsson_clang_release/lib/clang/5.0.0/include -I /usr/include/x86_64-linux-gnu -I /usr/include -D DEBUG -D _DEBUG -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/x86_64-linux-gnu/c++/6.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/x86_64-linux-gnu/c++/6.3.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/6.3.0/../../../../include/c++/6.3.0/backward -internal-isystem /usr/local/include -internal-isystem /home/gamesh411/ericsson_clang_debug/lib/clang/5.0.0/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -Wno-narrowing -Wno-delete-non-virtual-dtor -Wno-unnamed-type-template-args -Wno-comment -Wno-long-long -fdeprecated-macro -fdebug-compilation-dir /home/gamesh411/build_opencv3_git/opencvgcc/failed/histogram.cpp -ferror-limit 19 -fmessage-length 225 -fvisibility hidden -fvisibility-inlines-hidden -fobjc-runtime=gcc -fcxx-exceptions -fexceptions -fdiagnostics-show-option -fcolor-diagnostics -analyzer-opt-analyze-headers -analyzer-output=plist-multi-file -analyzer-disable-checker alpha.clone.CloneChecker -analyzer-disable-checker alpha.core.BoolAssignment -analyzer-disable-checker alpha.core.CallAndMessageUnInitRefArg -analyzer-disable-checker alpha.core.CastSize -analyzer-disable-checker alpha.core.CastToStruct -analyzer-disable-checker alpha.core.Conversion -analyzer-disable-checker alpha.core.DynamicTypeChecker -analyzer-disable-checker alpha.core.FixedAddr -analyzer-disable-checker alpha.core.IdenticalExpr -analyzer-disable-checker alpha.core.PointerArithm -analyzer-disable-checker alpha.core.PointerSub -analyzer-disable-checker alpha.core.SizeofPtr -analyzer-disable-checker alpha.core.TestAfterDivZero -analyzer-disable-checker alpha.cplusplus.IteratorRange -analyzer-disable-checker alpha.cplusplus.MisusedMovedObject -analyzer-disable-checker alpha.deadcode.UnreachableCode -analyzer-disable-checker alpha.osx.cocoa.DirectIvarAssignment -analyzer-disable-checker alpha.osx.cocoa.DirectIvarAssignmentForAnnotatedFunctions -analyzer-disable-checker alpha.osx.cocoa.InstanceVariableInvalidation -analyzer-disable-checker alpha.osx.cocoa.MissingInvalidationMethod -analyzer-disable-checker alpha.osx.cocoa.localizability.PluralMisuseChecker -analyzer-disable-checker alpha.security.ArrayBound -analyzer-disable-checker alpha.security.ArrayBoundV2 -analyzer-disable-checker alpha.security.MallocOverflow -analyzer-disable-checker alpha.security.ReturnPtrRange -analyzer-disable-checker alpha.security.taint.TaintPropagation -analyzer-disable-checker alpha.unix.BlockInCriticalSection -analyzer-disable-checker alpha.unix.Chroot -analyzer-disable-checker alpha.unix.PthreadLock -analyzer-disable-checker alpha.unix.SimpleStream -analyzer-disable-checker alpha.unix.Stream -analyzer-disable-checker alpha.unix.cstring.BufferOverlap -analyzer-disable-checker alpha.unix.cstring.NotNullTerminated -analyzer-disable-checker alpha.unix.cstring.OutOfBounds -analyzer-disable-checker apiModeling.google.GTest -analyzer-checker=core.CallAndMessage -analyzer-checker=core.DivideZero -analyzer-checker=core.DynamicTypePropagation -analyzer-checker=core.NonNullParamChecker -analyzer-checker=core.NullDereference -analyzer-checker=core.StackAddressEscape -analyzer-checker=core.UndefinedBinaryOperatorResult -analyzer-checker=core.VLASize -analyzer-checker=core.builtin.BuiltinFunctions -analyzer-checker=core.builtin.NoReturnFunctions -analyzer-checker=core.uninitialized.ArraySubscript -analyzer-checker=core.uninitialized.Assign -analyzer-checker=core.uninitialized.Branch -analyzer-checker=core.uninitialized.CapturedBlockVariable -analyzer-checker=core.uninitialized.UndefReturn -analyzer-checker=cplusplus.NewDelete -analyzer-checker=cplusplus.NewDeleteLeaks -analyzer-checker=cplusplus.SelfAssignment -analyzer-checker=deadcode.DeadStores -analyzer-disable-checker debug.AnalysisOrder -analyzer-disable-checker debug.ConfigDumper -analyzer-disable-checker debug.DumpBugHash -analyzer-disable-checker debug.DumpCFG -analyzer-disable-checker debug.DumpCallGraph -analyzer-disable-checker debug.DumpCalls -analyzer-disable-checker debug.DumpDominators -analyzer-disable-checker debug.DumpLiveVars -analyzer-disable-checker debug.DumpTraversal -analyzer-disable-checker debug.ExprInspection -analyzer-disable-checker debug.Stats -analyzer-disable-checker debug.TaintTest -analyzer-disable-checker debug.ViewCFG -analyzer-disable-checker debug.ViewCallGraph -analyzer-disable-checker debug.ViewExplodedGraph -analyzer-disable-checker llvm.Conventions -analyzer-disable-checker nullability.NullPassedToNonnull -analyzer-disable-checker nullability.NullReturnedFromNonnull -analyzer-disable-checker nullability.NullableDereferenced -analyzer-disable-checker nullability.NullablePassedToNonnull -analyzer-disable-checker nullability.NullableReturnedFromNonnull -analyzer-disable-checker optin.cplusplus.VirtualCall -analyzer-disable-checker optin.mpi.MPI-Checker -analyzer-disable-checker optin.osx.cocoa.localizability.EmptyLocalizationContextChecker -analyzer-disable-checker optin.osx.cocoa.localizability.NonLocalizedStringChecker -analyzer-disable-checker optin.performance.Padding -analyzer-disable-checker optin.portability.UnixAPI -analyzer-disable-checker osx.API -analyzer-disable-checker osx.NumberObjectConversion -analyzer-disable-checker osx.ObjCProperty -analyzer-disable-checker osx.SecKeychainAPI -analyzer-disable-checker osx.cocoa.AtSync -analyzer-disable-checker osx.cocoa.ClassRelease -analyzer-disable-checker osx.cocoa.Dealloc -analyzer-disable-checker osx.cocoa.IncompatibleMethodTypes -analyzer-disable-checker osx.cocoa.Loops -analyzer-disable-checker osx.cocoa.MissingSuperCall -analyzer-disable-checker osx.cocoa.NSAutoreleasePool -analyzer-disable-checker osx.cocoa.NSError -analyzer-disable-checker osx.cocoa.NilArg -analyzer-disable-checker osx.cocoa.NonNilReturnValue -analyzer-disable-checker osx.cocoa.ObjCGenerics -analyzer-disable-checker osx.cocoa.RetainCount -analyzer-disable-checker osx.cocoa.SelfInit -analyzer-disable-checker osx.cocoa.SuperDealloc -analyzer-disable-checker osx.cocoa.UnusedIvars -analyzer-disable-checker osx.cocoa.VariadicMethodTypes -analyzer-disable-checker osx.coreFoundation.CFError -analyzer-disable-checker osx.coreFoundation.CFNumber -analyzer-disable-checker osx.coreFoundation.CFRetainRelease -analyzer-disable-checker osx.coreFoundation.containers.OutOfBounds -analyzer-disable-checker osx.coreFoundation.containers.PointerSizedValues -analyzer-disable-checker security.FloatLoopCounter -analyzer-checker=security.insecureAPI.UncheckedReturn -analyzer-checker=security.insecureAPI.getpw -analyzer-checker=security.insecureAPI.gets -analyzer-checker=security.insecureAPI.mkstemp -analyzer-checker=security.insecureAPI.mktemp -analyzer-disable-checker security.insecureAPI.rand -analyzer-disable-checker security.insecureAPI.strcpy -analyzer-checker=security.insecureAPI.vfork -analyzer-checker=unix.API -analyzer-checker=unix.Malloc -analyzer-checker=unix.MallocSizeof -analyzer-checker=unix.MismatchedDeallocator -analyzer-checker=unix.StdCLibraryFunctions -analyzer-checker=unix.Vfork -analyzer-checker=unix.cstring.BadSizeArg -analyzer-checker=unix.cstring.NullArg -analyzer-disable-checker valist.CopyToSelf -analyzer-disable-checker valist.Uninitialized -analyzer-disable-checker valist.Unterminated -analyzer-config xtu-dir=/home/gamesh411/build_opencv3_git/opencvgcc/ctu-dir/x86_64 -o /home/gamesh411/build_opencv3_git/opencvgcc/histogram.cpp_5f1f94776187c199b4162da78b6f89f3.plist -x c++ /home/gamesh411/opencv-3.3.0/modules/imgproc/src/histogram.cpp 
1.      <eof> parser at end of file
2.      While analyzing stack: 
        #0 template<> inline explicit UMat::UMat<float>(const std::vector<float> &vec, bool copyData) : flags(MAGIC_VAL | DataType<float>::type | (1 << 14)), dims(2), rows((int)vec.size()), cols(1), allocator(0), usageFlags(USAGE_DEFAULT), u(0), offset(0), size(&this->rows), step() {}
3.      /home/gamesh411/opencv-3.3.0/modules/core/include/opencv2/core/mat.inl.hpp:3580:9: Error evaluating statement
4.      /home/gamesh411/opencv-3.3.0/modules/core/include/opencv2/core/mat.inl.hpp:3580:9: Error evaluating statement
clang-5.0: error: unable to execute command: Aborted (core dumped)
clang-5.0: error: clang frontend command failed due to signal (use -v to see invocation)
clang version 5.0.0 (https://github.com/Ericsson/clang.git 12db6330258c8209fc19f56412e9f1a63f560d2c) (http://llvm.org/git/llvm.git 657c31173ea30090583e40c7a9204561d9c2d8c4)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /home/gamesh411/ericsson_clang_debug/bin
clang-5.0: note: diagnostic msg: PLEASE submit a bug report to http://llvm.org/bugs/ and include the crash backtrace, preprocessed source, and associated run script.
clang-5.0: note: diagnostic msg: 
********************

PLEASE ATTACH THE FOLLOWING FILES TO THE BUG REPORT:
Preprocessed source(s) and associated run script(s) are located at:
clang-5.0: note: diagnostic msg: /tmp/histogram-580980.cpp
clang-5.0: note: diagnostic msg: /tmp/histogram-580980.sh
clang-5.0: note: diagnostic msg: 

********************

```

--- a/lib/AST/ASTImporter.cpp
+++ b/lib/AST/ASTImporter.cpp
@@ -141,6 +141,7 @@ namespace clang {
                            bool Complain = true);
     bool IsStructuralMatch(EnumDecl *FromEnum, EnumDecl *ToRecord,
                            bool Complain = true);
+    bool IsStructuralMatch(FunctionDecl *From, FunctionDecl *To);
     bool IsStructuralMatch(FunctionTemplateDecl *From, FunctionTemplateDecl *To);
     bool IsStructuralMatch(EnumConstantDecl *FromEC, EnumConstantDecl *ToEC);
     bool IsStructuralMatch(ClassTemplateDecl *From, ClassTemplateDecl *To);
@@ -1357,6 +1358,13 @@ bool ASTNodeImporter::IsStructuralMatch(FunctionTemplateDecl *From,
   return Ctx.IsStructurallyEquivalent(From, To);
 }
 
+bool ASTNodeImporter::IsStructuralMatch(FunctionDecl *From, FunctionDecl *To) {
+  StructuralEquivalenceContext Ctx(
+      Importer.getFromContext(), Importer.getToContext(),
+      Importer.getNonEquivalentDecls(), false, false);
+  return Ctx.IsStructurallyEquivalent(From, To);
+}
+
 bool ASTNodeImporter::IsStructuralMatch(EnumConstantDecl *FromEC,
                                         EnumConstantDecl *ToEC)
 {
@@ -2140,8 +2148,7 @@ Decl *ASTNodeImporter::VisitFunctionDecl(FunctionDecl *D) {
       if (FunctionDecl *FoundFunction = dyn_cast<FunctionDecl>(FoundDecls[I])) {
         if (FoundFunction->hasExternalFormalLinkage() &&
             D->hasExternalFormalLinkage()) {
-          if (Importer.IsStructurallyEquivalent(D->getType(), 
-                                                FoundFunction->getType())) {
+          if (IsStructuralMatch(D, FoundFunction)) {
             // FIXME: Actually try to merge the body and other attributes.
             const FunctionDecl *FromBodyDecl = nullptr;
             D->hasBody(FromBodyDecl);

--- a/lib/AST/ASTImporter.cpp
+++ b/lib/AST/ASTImporter.cpp
@@ -4995,8 +4995,8 @@ Expr *ASTNodeImporter::VisitDeclRefExpr(DeclRefExpr *E) {
   if (T.isNull())
     return nullptr;
 
-
-  TemplateArgumentListInfo ToTAInfo;
+  TemplateArgumentListInfo ToTAInfo(Importer.Import(E->getLAngleLoc()),
+                                    Importer.Import(E->getRAngleLoc()));
   TemplateArgumentListInfo *ResInfo = nullptr;
   if (E->hasExplicitTemplateArgs()) {
     for (const auto &FromLoc : E->template_arguments()) {
@@ -5929,18 +5929,25 @@ Expr *ASTNodeImporter::VisitMemberExpr(MemberExpr *E) {
     Importer.Import(E->getMemberNameInfo().getName()),
     Importer.Import(E->getMemberNameInfo().getLoc()));
 
+  TemplateArgumentListInfo ToTAInfo(Importer.Import(E->getLAngleLoc()),
+                                    Importer.Import(E->getRAngleLoc()));
+  TemplateArgumentListInfo *ResInfo = nullptr;
   if (E->hasExplicitTemplateArgs()) {
-    return nullptr; // FIXME: handle template arguments
+    for (const auto &FromLoc : E->template_arguments()) {
+      if (auto ToTALoc = ImportTemplateArgumentLoc(FromLoc)) {
+        ToTAInfo.addArgument(*ToTALoc);
+      } else
+        return nullptr;
+    }
+    ResInfo = &ToTAInfo;
   }
 
-  return MemberExpr::Create(Importer.getToContext(), ToBase,
-                            E->isArrow(),
+  return MemberExpr::Create(Importer.getToContext(), ToBase, E->isArrow(),
                             Importer.Import(E->getOperatorLoc()),
                             Importer.Import(E->getQualifierLoc()),
                             Importer.Import(E->getTemplateKeywordLoc()),
-                            ToMember, ToFoundDecl, ToMemberNameInfo,
-                            nullptr, T, E->getValueKind(),
-                            E->getObjectKind());
+                            ToMember, ToFoundDecl, ToMemberNameInfo, ResInfo, T,
+                            E->getValueKind(), E->getObjectKind());
 }
 
 Expr *ASTNodeImporter::VisitCXXDependentScopeMemberExpr(
@@ -5954,7 +5961,8 @@ Expr *ASTNodeImporter::VisitCXXDependentScopeMemberExpr(
   if (!E->getBaseType().isNull() && BaseType.isNull())
     return nullptr;
 
-  TemplateArgumentListInfo ToTAInfo;
+  TemplateArgumentListInfo ToTAInfo(Importer.Import(E->getLAngleLoc()),
+                                    Importer.Import(E->getRAngleLoc()));
   TemplateArgumentListInfo *ResInfo = nullptr;
   if (E->hasExplicitTemplateArgs()) {
     for (const auto &FromLoc : E->template_arguments()) {
@@ -5994,7 +6002,8 @@ ASTNodeImporter::VisitDependentScopeDeclRefExpr(DependentScopeDeclRefExpr *E) {
 
   ImportDeclarationNameLoc(E->getNameInfo(), NameInfo);
 
-  TemplateArgumentListInfo ToTAInfo;
+  TemplateArgumentListInfo ToTAInfo(Importer.Import(E->getLAngleLoc()),
+                                    Importer.Import(E->getRAngleLoc()));
   TemplateArgumentListInfo *ResInfo = nullptr;
   if (E->hasExplicitTemplateArgs()) {
     for (const auto &FromLoc : E->template_arguments()) {
@@ -6060,7 +6069,8 @@ Expr *ASTNodeImporter::VisitUnresolvedLookupExpr(UnresolvedLookupExpr *E) {
       return nullptr;
   }
 
-  TemplateArgumentListInfo ToTAInfo;
+  TemplateArgumentListInfo ToTAInfo(Importer.Import(E->getLAngleLoc()),
+                                    Importer.Import(E->getRAngleLoc()));
   TemplateArgumentListInfo *ResInfo = nullptr;
   if (E->hasExplicitTemplateArgs()) {
     for (const auto &FromLoc : E->template_arguments()) {

--- a/lib/AST/ASTImporter.cpp
+++ b/lib/AST/ASTImporter.cpp
@@ -2321,6 +2321,12 @@ Decl *ASTNodeImporter::VisitFunctionDecl(FunctionDecl *D) {
     ToFunction->setType(T);
   }
 
+  // Import the describing template function, if any.
+  if (FunctionTemplateDecl *FromFT = D->getDescribedFunctionTemplate())
+    if (auto *ToFT = dyn_cast<FunctionTemplateDecl>(Importer.Import(FromFT))) {
+      ToFunction->setDescribedFunctionTemplate(ToFT);
+    }
+
   // Import the body, if any.
   if (Stmt *FromBody = D->getBody()) {
     if (Stmt *ToBody = Importer.Import(FromBody)) {
@@ -4325,6 +4331,11 @@ Decl *ASTNodeImporter::VisitFunctionTemplateDecl(FunctionTemplateDecl *D) {
   if (ToD)
       return ToD;
 
+  FunctionDecl *TemplatedFD =
+      cast_or_null<FunctionDecl>(Importer.Import(D->getTemplatedDecl()));
+  if (!TemplatedFD)
+    return nullptr;
+
   // Try to find a function in our own ("to") context with the same name, same
   // type, and in the same context as the function we're importing.
   if (!LexicalDC->isFunctionOrMethod()) {
@@ -4352,11 +4363,6 @@ Decl *ASTNodeImporter::VisitFunctionTemplateDecl(FunctionTemplateDecl *D) {
   TemplateParameterList *Params = ImportTemplateParameterList(
                                     D->getTemplateParameters());
   if (!Params)
-    return nullptr;
-
-  FunctionDecl *TemplatedFD = cast_or_null<FunctionDecl>(
-		                 Importer.Import(D->getTemplatedDecl()));
-  if (!TemplatedFD)
     return nullptr;
 
   FunctionTemplateDecl *ToFunc = FunctionTemplateDecl::Create(

--- a/lib/AST/ASTStructuralEquivalence.cpp
+++ b/lib/AST/ASTStructuralEquivalence.cpp
@@ -1210,6 +1210,45 @@ static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
   if (!::IsStructurallyEquivalent(Context, D1->getType(), D2->getType()))
     return false;
 
+  if (D1->getDescribedFunctionTemplate()) {
+    if (D2->getDescribedFunctionTemplate()) {
+      if (!Context.IsStructurallyEquivalent(D1->getDescribedFunctionTemplate(),
+                                            D2->getDescribedFunctionTemplate()))
+        return false;
+    } else {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+static bool IsStructurallyEquivalent(StructuralEquivalenceContext &Context,
+                                     FunctionTemplateDecl *D1,
+                                     FunctionTemplateDecl *D2) {
+  if (!Context.IsStructurallyEquivalent(D1->getTemplatedDecl(),
+                                        D2->getTemplatedDecl()))
+    return false;
+
+  // Check the template parameter list.
+  TemplateParameterList *TParams1 = D1->getTemplateParameters();
+  TemplateParameterList *TParams2 = D2->getTemplateParameters();
+
+  TemplateParameterList::iterator TP2 = TParams2->begin(),
+                                  TP2End = TParams2->end();
+  for (TemplateParameterList::iterator TP1 = TParams1->begin(),
+                                       TP1End = TParams1->end();
+       TP1 != TP1End; ++TP1, ++TP2) {
+    if (TP2 == TP2End)
+      return false;
+
+    if (!IsStructurallyEquivalent(Context, *TP1, *TP2))
+      return false;
+  }
+
+  if (TP2 != TP2End)
+    return false;
+
   return true;
 }
 
@@ -1431,6 +1470,17 @@ bool StructuralEquivalenceContext::Finish() {
           // Kind mismatch.
           Equivalent = false;
         }
+      }
+    } else if (FunctionTemplateDecl *FD1 = dyn_cast<FunctionTemplateDecl>(D1)) {
+      if (FunctionTemplateDecl *FD2 = dyn_cast<FunctionTemplateDecl>(D2)) {
+        if (!::IsStructurallyEquivalent(FD1->getIdentifier(),
+                                        FD2->getIdentifier()))
+          Equivalent = false;
+        else if (!::IsStructurallyEquivalent(*this, FD1, FD2))
+          Equivalent = false;
+      } else {
+        // Kind mismatch.
+        Equivalent = false;
       }
     }
 

--- a/unittests/AST/ASTImporterTest.cpp
+++ b/unittests/AST/ASTImporterTest.cpp
@@ -495,6 +495,22 @@ TEST(ImportDecl, ImportFunctionTemplateDecl) {
           "template <typename T> void declToImport() { };",
           Lang_CXX, "", Lang_CXX, Verifier,
           functionTemplateDecl()));
+  EXPECT_TRUE(testImport(
+      "template<typename Y> int a() { return 1; }"
+      "template<typename Y, typename D> int a(){ return 2; }"
+      "void declToImport() { a<void>(); }",
+      Lang_CXX, "", Lang_CXX, Verifier,
+      functionDecl(has(compoundStmt(has(callExpr(has(ignoringParenImpCasts(
+          declRefExpr(to(functionDecl(hasBody(compoundStmt(
+              has(returnStmt(has(integerLiteral(equals(1)))))))))))))))))));
+  EXPECT_TRUE(testImport(
+      "template<typename Y> int a() { return 1; }"
+      "template<typename Y, typename D> int a() { return 2; }"
+      "void declToImport() { a<void,void>(); }",
+      Lang_CXX, "", Lang_CXX, Verifier,
+      functionDecl(has(compoundStmt(has(callExpr(has(ignoringParenImpCasts(
+          declRefExpr(to(functionDecl(hasBody(compoundStmt(
+              has(returnStmt(has(integerLiteral(equals(2)))))))))))))))))));
 }
 
 TEST(ImportExpr, ImportUnresolvedLookupExpr) {

